unyt/mpl_interface.py

```diff
@@ -18,12 +18,14 @@
 except ImportError:
     pass
 else:
+    from weakref import WeakKeyDictionary
     from unyt import unyt_array, unyt_quantity, Unit
 
     class unyt_arrayConverter(ConversionInterface):
         """Matplotlib interface for unyt_array"""
 
         _labelstyle = "()"
+        _axisnames = WeakKeyDictionary()
```

Comment:
What if someone makes a multipanel plot with two x-axes. Won't the second x-axis clobber the first one?

In general having mutable state defined at class level like this is usually not what you want. Since this class normally acts as a singleton that's probably actually OK here, but then you need to address that this is a global key-value store and find a way to make the keys unique. Right now I don't see how you're enforcing uniqueness in the keys.

Reply:
> What if someone makes a multipanel plot with two x-axes. Won't the second x-axis clobber the first one?

Matplotlib will generate two unique XAxis instances that are passed to our conversion interface. So, it won't clobber the label. I added a test for multiple subplots.



> Right now I don't see how you're enforcing uniqueness in the keys.

The Axis object is hashable. I have added a test for this. Is hashable sufficient?

Reply:
Oops, thought you were using the name, sorry for misreading.

I'm still torn on this and want to wait until next week to come to a decision.